crates/biome_js_analyze/src/lint/nursery/no_excessive_lines_per_function.rs

```diff
@@ -0,0 +1,286 @@
+use crate::services::semantic::Semantic;
+use ::serde::{Deserialize, Serialize};
+use biome_analyze::{
+    Rule, RuleDiagnostic, RuleSource, RuleSourceKind, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{
+    AnyFunctionLike, AnyJsFunction, JsCallExpression, JsParenthesizedExpression,
+};
+use biome_rowan::AstNode;
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+use std::num::NonZeroU8;
+
+declare_lint_rule! {
+    /// Restrict the number of lines of code in a function.
+    ///
+    /// This rule checks the number of lines in a function body and reports a diagnostic if it exceeds a specified limit.
+    /// Some people consider large functions a code smell. Large functions tend to do a lot of things and can make it hard following what’s going on. Many coding style guides dictate a limit of the number of lines that a function can comprise of. This rule can help enforce that style.
+    ///
+    /// ## Examples
+    ///
+    /// When maximum number of lines is set to `2` with the following option, the following code will be considered invalid:
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///        "maxLines": 2
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// function foo () {
+    ///   const x = 0;
+    ///   const y = 1;
+    ///   const z = 2;
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// const bar = () => {
+    ///   const x = 0;
+    ///   const y = 1;
+    ///   const z = 2;
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// class Baz {
+    ///	  foo() {
+    ///     const x = 0;
+    ///	    const y = 0;
+    ///     const z = 2;
+    ///	  };
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// (() => {
+    ///	 const x = 0;
+    ///	 const y = 0;
+    ///	 const z = 0;
+    /// })();
+    /// ```
```

Comment:
FYI, these examples don't need to be as comprehensive as unit tests. The purpose of the examples is to help explain the rule to the user visually. However, too many examples that aren't different enough are just visual noise.

IMO, I think a single `function foo() { ... }` example is enough to get the point across.